src/libraries/System.Memory/tests/Span/Reverse.cs

```diff
@@ -70,19 +73,37 @@ public static void ReverseByteTwiceReturnsOriginal()
             Assert.Equal<byte>(expected, actual);
         }
 
-        [Fact]
-        public static void ReverseByteUnaligned()
+        public static IEnumerable<object[]> GetReverseByteUnalignedArguments()
         {
-            const int length = 32;
             const int offset = 1;
+
+            yield return new object[] { offset, 4 };
+            yield return new object[] { offset, 5 };
+
+            // vectorized execution paths
+            if (Avx2.IsSupported)
+            {
+                yield return new object[] { offset, offset + Vector256<byte>.Count * 2 }; // even
+                yield return new object[] { offset, offset + Vector256<byte>.Count * 2 + 1 }; // odd
```

Comment:
Why conditionalize these tests on whether Avx2.IsSupported rather than just always testing these sizes?  Same for Sse2.IsSupported below.